pageserver/src/tenant.rs

```diff
@@ -271,10 +271,7 @@ impl UninitializedTimeline<'_> {
             .await
             .context("Failed to flush after basebackup import")?;
 
-        // Initialize without loading the layer map. We started with an empty layer map, and already
-        // updated it for the layers that we created during the import.
-        let mut timelines = self.owning_tenant.timelines.lock().unwrap();
-        self.initialize_with_lock(ctx, &mut timelines, false, true)
```

Comment:
I have already lost the meaning of this, it was a bit of extra on the original PR, correctness related still.

Reply:
Original explanation from https://github.com/neondatabase/neon/pull/3869#discussion_r1159950179:

> As I noticed several times this PR is not just adding check for duplicates.
> This check actually detects errors.
> One of them is related with importing file fro backup. Right now it cause loading each layer twice.
> I have to pass load_timeline=false to initialize_with_lock to prevent duplication.

Reply:
This part seems fine to me. Probably best to revert it anyway, if we're not 100% that it's orthogonal to the other changes.

Reply:
Ok, sorry for not seeing your comment... Seeing them out of order actually.

Reply:
Ok, keeping this as is.

Reply:
This piece, as well as the comment added in `bootstrap_timeline`, seem orthogonal to me, but, also safe to revert. Let's re-introduce them as an independent change after the release.